mathsnuggets/widgets/change_subject.py

```diff
@@ -0,0 +1,21 @@
+import sympy
+
+from mathsnuggets.core import fields, form
+
+test = {"equation": "y=mx+c"}
+
+
+class ChangeSubject(form.Form):
+    """Change the subject"""
+
+    template = "Make `variable` the subject for `equation` `solution`"
+
+    equation = fields.Equation("Equation", required=True)
+    variable = fields.Expression("Solve for", default="x", required=True)
+
+    @fields.computed("Solution")
+    def solution(self):
+        solution = sympy.solve(self.equation, self.variable)
+        if not solution:
+            raise ValueError("The equation you have entered cannot be solved")
```

Comment:
Speficy which equation you're talking about. See repr in python and how it's used in other parts of the code